tests/run-make/obey-crate-type-flag/rmake.rs

```diff
@@ -0,0 +1,18 @@
+// test.rs should produce both an rlib and a dylib
+// by default. When the crate_type flag is passed and
+// forced to dylib, no rlibs should be produced.
+// See https://github.com/rust-lang/rust/issues/11573
+
+//@ ignore-cross-compile
+
+use run_make_support::{
+    cwd, dynamic_lib_name, fs_wrapper, has_extension, rust_lib_name, rustc, shallow_find_files,
+};
+
+fn main() {
+    rustc().input("test.rs").run();
+    fs_wrapper::remove_file(dynamic_lib_name("test"));
+    fs_wrapper::remove_file(rust_lib_name("test"));
```

Comment:
Suggestion: ditto here about checking for existence, not checking-by-proxy via trying to remove file.